python/common/org/python/stdlib/datetime/DateTime.java

```diff
@@ -0,0 +1,377 @@
+	public org.python.Object timestamp() {
+
+		int year = (int)this.timeUnits[YEAR_INDEX];
+
+		int month = (int)this.timeUnits[MONTH_INDEX];
+
+		int day = (int)this.timeUnits[DAY_INDEX];
+
+		int hour = (int)this.timeUnits[HOUR_INDEX];
+
+		int minute = (int)this.timeUnits[MINUTE_INDEX];
+
+		int microsec = (int)this.timeUnits[MICROSECOND_INDEX];
+		
+		int seconds = (int)this.timeUnits[SECOND_INDEX];
+
+		Calendar now = new java.util.GregorianCalendar(year,month - 1,day,hour,minute,seconds);
+		now.set(Calendar.MILLISECOND, microsec/1000);
+
+		double totalsec = now.getTimeInMillis() / 1000;
+		return new org.python.types.Float(totalsec);
+
+	}
+    
+	// Added class method
+
+	@org.python.Method(__doc__ = "Returns a datetime object based on a iso-formatted input")
+	public static org.python.Object fromisoformat(String isoString) {
+
+	SimpleDateFormat sdf = new java.text.SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSSSS");
+
+	java.util.Date isodt;
+	try {
+		isodt = sdf.parse(isoString); // create date obj based on iso string
+	} catch (ParseException pe) {
+		System.out.println(pe);
+		throw new org.python.exceptions.ValueError("Invalid isoformat string: " + isoString);
+	}
+
+	SimpleDateFormat sdfYear = new SimpleDateFormat("yyyy");
+	SimpleDateFormat sdfMonth = new SimpleDateFormat("MM");
+	SimpleDateFormat sdfDay = new SimpleDateFormat("dd");
+	SimpleDateFormat sdfHour = new SimpleDateFormat("HH");
+	SimpleDateFormat sdfMin = new SimpleDateFormat("mm");
+	SimpleDateFormat sdfSec = new SimpleDateFormat("ss");
+	SimpleDateFormat sdfMic = new SimpleDateFormat("SSSSSS");
+
+	String getYear = sdfYear.format(isodt);
+	String getMonth = sdfMonth.format(isodt);
+	String getDay = sdfDay.format(isodt);
+	String getHour = sdfHour.format(isodt);
+	String getMin = sdfMin.format(isodt);
+	String getSec = sdfSec.format(isodt);
+	String getMic = sdfMic.format(isodt);
+
+	int iYear = Integer.parseInt(getYear);
+	int iMonth = Integer.parseInt(getMonth);
+	int iDay = Integer.parseInt(getDay);
+	int iHour = Integer.parseInt(getHour);
+	int iMin = Integer.parseInt(getMin);
+	int iSec = Integer.parseInt(getSec);
+	int iMic = Integer.parseInt(getMic);
+	
+	org.python.types.Int yea = org.python.types.Int.getInt(iYear);
+	org.python.types.Int mont = org.python.types.Int.getInt(iMonth);
+	org.python.types.Int da = org.python.types.Int.getInt(iDay);
+	org.python.types.Int hou = org.python.types.Int.getInt(iHour);
+	org.python.types.Int minut = org.python.types.Int.getInt(iMin);
+	org.python.types.Int secon = org.python.types.Int.getInt(iSec);
+	org.python.types.Int microsecon = org.python.types.Int.getInt(iMic); 
+
+	org.python.Object[] args = { yea, mont, da, hou, minut, secon, microsecon };
+	return new DateTime(args, Collections.emptyMap());
+
+	}
+
+	@org.python.Method(__doc__ = "Checks if current dateobject is less than dt")
```

Comment:
Nicely done with the comparision-methods!

Reply:
ty